Yona/Yona/Notifications/NotificationsViewController.swift

```diff
@@ -377,24 +377,16 @@ class NotificationsViewController: UITableViewController, YonaUserSwipeCellDeleg
     }
     
     func showBuddyProfile(_ theMessage : Message) {
-        UserRequestManager.sharedInstance.getUser(GetUserRequest.notAllowed, onCompletion: {(succes, serverMessage, serverCode, aUser) in
-            var countPush = 0
-            if succes {
-                if let user = aUser {
-                    for aBuddies in user.buddies {
-                        if aBuddies.UserRequestSelfLink == theMessage.UserRequestSelfLink {
-                            let storyBoard: UIStoryboard = UIStoryboard(name:"Friends", bundle: Bundle.main)
-                            let controller = storyBoard.instantiateViewController(withIdentifier: "FriendsProfileViewController") as! FriendsProfileViewController
-                            controller.aUser = aBuddies
-                            //Make sure it only pushes once!
-                            if countPush <= 0 {
-                                self.navigationController?.pushViewController(controller, animated: true)
-                            }
-                            countPush += 1
-                        }
-                    }
-                }
-            }
-        })
+        guard let savedUser = UserDefaults.standard.object(forKey: YonaConstants.nsUserDefaultsKeys.savedUser) else {
+            return
+        }
+        let user = UserRequestManager.sharedInstance.convertToDictionary(text: savedUser as! String)
+        let newUser = Users.init(userData: user! as BodyDataDictionary)
+        if let aBuddies = newUser.buddies.first(where: { $0.UserRequestSelfLink == theMessage.yonaUserLink }) {
```

Comment:
The variable name ``aBuddies`` is wrong, as this is just a single buddy.  Please make it just ``buddy``.

Reply:
Updated.